service/controller/cluster.go

```diff
@@ -569,10 +571,10 @@ func newClusterResources(config ClusterConfig, certsSearcher certs.Interface) ([
 		namespaceResource,
 		serviceResource,
 		resourceGroupResource,
-		containerURLResource,
 		encryptionkeyResource,
-		blobObjectResource,
 		deploymentResource,
+		containerURLResource,
+		blobObjectResource,
```

Comment:
What affects this resource ordering? Changes in `deploymentResource` seemed like contained. Is this needed?

Reply:
yes, because if the public IP address of the control plane changes, the containerURLResource and blobObjectResource will fail and stop the reconciliation.
The deploymentResource has to run before them to update the storage account to the newest IP addresses otherwise we get a crash loop.

Reply:
Ah, right, and it's the `instanceResource` then what further uses the blob objects uploaded by `blobObjectResource`?

Reply:
yes, together with the `masterResource`

Reply:
Coolio!